src/menu.rs

```diff
@@ -6,36 +6,98 @@ use std::{
   hash::{Hash, Hasher},
 };
 
+use crate::platform_impl::{CustomMenuItem as CustomMenuItemPlatform, Menu as MenuPlatform};
+
+/// Tray menu (Also known as context menu)
+pub struct Tray;
+/// Menubar menu (It represent the Window menu for Windows and Linux)
+pub struct Menubar;
+
+impl Tray {
+  /// Creates a new Tray (Context) menu for platforms where this is appropriate.
+  ///
+  /// This function is equivalent to [`Menu::new_popup_menu(MenuType::SystemTray)`].
+  /// [`Menu::new_popup_menu(MenuType::SystemTray)`]: crate::menu::Menu
+  #[allow(clippy::new_ret_no_self)]
+  pub fn new() -> Menu {
+    Menu {
+      menu_platform: MenuPlatform::new_popup_menu(),
+      menu_type: MenuType::SystemTray,
+    }
+  }
+}
+
+impl Menubar {
+  /// Creates a new Menubar (Window) menu for platforms where this is appropriate.
+  ///
+  /// This function is equivalent to [`Menu::new(MenuType::Menubar)`].
+  /// [`Menu::new_popup_menu(MenuType::Menubar)`]: crate::menu::Menu
+  #[allow(clippy::new_ret_no_self)]
+  pub fn new() -> Menu {
+    Menu {
+      menu_platform: MenuPlatform::new(),
+      menu_type: MenuType::Menubar,
+    }
+  }
+}
+
+/// Base `Menu` functions.
 #[derive(Debug, Clone)]
 pub struct Menu {
-  pub title: String,
-  pub items: Vec<MenuItem>,
+  pub(crate) menu_platform: MenuPlatform,
+  pub(crate) menu_type: MenuType,
 }
 
 impl Menu {
-  pub fn new(title: &str, items: Vec<MenuItem>) -> Self {
+  /// Creates a new Menu for Menubar/Window context.
+  pub fn new(menu_type: MenuType) -> Self {
     Self {
-      title: String::from(title),
-      items,
+      menu_platform: MenuPlatform::new(),
+      menu_type,
     }
   }
-}
 
-#[derive(Debug, Clone, Hash)]
-/// CustomMenu is a custom menu who emit an event inside the EventLoop.
-pub struct CustomMenu {
-  pub id: MenuId,
-  pub name: String,
-  pub keyboard_accelerators: Option<String>,
+  /// Creates a new Menu for Popup context.
+  pub fn new_popup_menu(menu_type: MenuType) -> Self {
+    Self {
+      menu_platform: MenuPlatform::new_popup_menu(),
+      menu_type,
+    }
+  }
+
+  /// Shortcut to add a submenu to this `Menu`.
+  ///
+  /// This function is equivalent to [`Menu::add_item(self, MenuItem::Submenu(title, enabled, submenu))`].
```

Comment:
This won't get the link
```suggestion
  /// This function is equivalent to `Menu::add_item(self, MenuItem::Submenu(title, enabled, submenu))`.
```